plugins/funind/functional_principles_types.ml

```diff
@@ -320,7 +319,7 @@ let build_functional_principle (evd:Evd.evar_map ref) interactive_proof old_prin
     (* 	  let dur1 = System.time_difference tim1 tim2 in *)
     (* 	  Pp.msgnl (str ("Time to compute proof: ") ++ str (string_of_float dur1)); *)
     (* 	end; *)
-    get_proof_clean true, CEphemeron.create hook
+    get_proof_clean true, hook
```

Comment:
If there was a ephemeron, wasn't there a reason for it?

Reply:
As of today hooks are captured in the `terminators` thus already wrapped in an "ephemeron" by `Proof_global`. So indeed I dont' see why this is necessary, but I agree that a more careful review is needed.

Reply:
> As of today hooks are captured in the terminators thus already wrapped in an "ephemeron" by Proof_global

This is the case in `start_proof` but here, the hook is passed to `Indfun_common.save` which does no encapsulation in a terminator.

I mean, I don't undertand well what the impact of using ephemeron or not is but this precise hook is not encapulated in a terminator.

Reply:
Indeed thanks, let me investigate more.